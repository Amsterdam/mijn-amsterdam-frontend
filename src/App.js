import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import { AppRoutes } from './App.constants.js';
import MainHeader from './components/MainHeader/MainHeader.js';
import MainFooter from './components/MainFooter/MainFooter.js';
import Dashboard from './pages/Dashboard/Dashboard.js';
import Profile from './pages/Profile/Profile.js';
import NotFound from './pages/NotFound/NotFound.js';
import styles from './App.module.scss';

import AppContext from './App.context';
import { useBrpApi } from 'hooks/brp-api.hook.js';

export default function App() {
  // State that needs to be available everywhere
  const appState = {
    BRP: useBrpApi(),
  };

  return (
    <BrowserRouter>
      <AppContext.Provider value={appState}>
        <MainHeader />
        <main className={styles.App}>
          <Switch>
            <Route exact path={AppRoutes.ROOT} component={Dashboard} />
            {/* <Route path={AppRoutes.BURGERZAKEN} />
              <Route path={AppRoutes.WONEN} />
              <Route path={AppRoutes.BELASTINGEN} />
              <Route path={AppRoutes.GEZONDHEID} />
              <Route path={AppRoutes.INKOMEN} />*/}
            <Route path={AppRoutes.PROFILE} component={Profile} />
            <Route component={NotFound} />
          </Switch>
        </main>
        <MainFooter />
      </AppContext.Provider>
    </BrowserRouter>
  );
}
