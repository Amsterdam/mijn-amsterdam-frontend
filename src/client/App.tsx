import * as Sentry from '@sentry/browser';
import classnames from 'classnames';
import React, { useContext, useEffect } from 'react';
import ErrorBoundary from 'react-error-boundary';
import {
  BrowserRouter,
  Redirect,
  Route,
  Switch,
  useLocation,
} from 'react-router-dom';
import { AppRoutes, FeatureToggle } from '../universal/config';
import { getOtapEnvItem, IS_PRODUCTION } from '../universal/config/env';
import { isPrivateRoute } from '../universal/helpers';
import styles from './App.module.scss';
import {
  ApplicationError,
  AutoLogoutDialog,
  MainFooter,
  MainHeader,
} from './components';
import { DefaultAutologoutDialogSettings } from './components/AutoLogoutDialog/AutoLogoutDialog';
import {
  TMA_LOGIN_URL_DIGID_AFTER_REDIRECT,
  TMA_LOGIN_URL_EHERKENNING_AFTER_REDIRECT,
} from './config/api';
import {
  useAnalytics,
  useLocalStorage,
  usePageChange,
  useScript,
} from './hooks';
import {
  Burgerzaken,
  BurgerzakenDetail,
  Dashboard,
  GarbageInformation,
  Inkomen,
  InkomenDetail,
  InkomenDetailTozo,
  InkomenSpecificaties,
  LandingPage,
  MyArea,
  MyNotifications,
  MyTips,
  NotFound,
  Profile,
  Zorg,
  ZorgDetail,
  GeneralInfo,
  Vergunningen,
  VergunningDetail,
  Accessibility,
} from './pages';
import { SessionContext, SessionState } from './SessionState';
import { RecoilRoot } from 'recoil';
import ProfileCommercial from './pages/Profile/ProfileCommercial';
import { useAppState } from './hooks/useAppState';

function AppNotAuthenticated() {
  const location = useLocation();

  const [routeEntry, setRouteEntry] = useLocalStorage('RouteEntry', '');

  if (
    (!routeEntry || routeEntry === '/') &&
    location.pathname !== '/' &&
    isPrivateRoute(location.pathname)
  ) {
    setRouteEntry(location.pathname);
  }

  return (
    <>
      <div className={classnames(styles.App, styles.NotYetAuthenticated)}>
        <MainHeader />
        <Switch>
          <Route exact path={AppRoutes.ROOT} component={LandingPage} />
          <Route path={AppRoutes.ACCESSIBILITY} component={Accessibility} />
          <Route
            render={({ location: { pathname } }) => {
              if (isPrivateRoute(pathname)) {
                // Private routes are redirected to Home
                return <Redirect to={AppRoutes.ROOT} />;
              }
              // All other routes are presented with a 404 page
              return <Route component={NotFound} />;
            }}
          />
        </Switch>
      </div>
      <MainFooter />
    </>
  );
}

function AppAuthenticated() {
  useAppState();
  const location = useLocation();
  const session = useContext(SessionContext);
  const [routeEntry, setRouteEntry] = useLocalStorage('RouteEntry', '');

  const redirectAfterLogin = routeEntry || AppRoutes.ROOT;

  usePageChange();

  useEffect(() => {
    if (routeEntry) {
      setRouteEntry('');
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return location.pathname === AppRoutes.BUURT ? (
    <MyArea />
  ) : (
    <>
      <MainHeader isAuthenticated={session.isAuthenticated} />
      <div className={styles.App} id="AppContent">
        <Switch>
          <Redirect
            from={TMA_LOGIN_URL_DIGID_AFTER_REDIRECT}
            to={redirectAfterLogin}
          />
          <Redirect
            from={TMA_LOGIN_URL_EHERKENNING_AFTER_REDIRECT}
            to={redirectAfterLogin}
          />
          <Route exact path={AppRoutes.ROOT} component={Dashboard} />
          <Route path={AppRoutes.NOTIFICATIONS} component={MyNotifications} />
          <Route path={AppRoutes.BRP} component={Profile} />
          <Route path={AppRoutes.KVK} component={ProfileCommercial} />
          <Route path={AppRoutes.TIPS} component={MyTips} />
          <Route
            path={AppRoutes['INKOMEN/STADSPAS']}
            component={InkomenDetail}
          />
          <Route
            path={AppRoutes['INKOMEN/BIJSTANDSUITKERING']}
            component={InkomenDetail}
          />
          <Route
            path={AppRoutes['INKOMEN/SPECIFICATIES']}
            component={InkomenSpecificaties}
          />
          {FeatureToggle.tozoActive && (
            <Route
              path={AppRoutes['INKOMEN/TOZO']}
              component={InkomenDetailTozo}
            />
          )}
          <Route path={AppRoutes.INKOMEN} component={Inkomen} />
          <Route
            path={AppRoutes['ZORG/VOORZIENINGEN']}
            component={ZorgDetail}
          />
          <Route path={AppRoutes.ZORG} component={Zorg} />
          <Route
            path={AppRoutes.BURGERZAKEN_DOCUMENT}
            component={BurgerzakenDetail}
          />
          <Route path={AppRoutes.BURGERZAKEN} component={Burgerzaken} />
          {FeatureToggle.garbageInformationPage && (
            <Route path={AppRoutes.AFVAL} component={GarbageInformation} />
          )}
          <Route path={AppRoutes.ACCESSIBILITY} component={Accessibility} />
          <Route path={AppRoutes.GENERAL_INFO} component={GeneralInfo} />
          <Route
            path={AppRoutes['VERGUNNINGEN/DETAIL']}
            component={VergunningDetail}
          />
          <Route path={AppRoutes.VERGUNNINGEN} component={Vergunningen} />
          <Route component={NotFound} />
        </Switch>
      </div>
      <MainFooter />
    </>
  );
}

function AppLanding() {
  const session = useContext(SessionContext);

  const { isPristine, isAuthenticated, validityInSeconds } = session;

  // If session was previously authenticated we don't want to show the loader again
  if (isPristine) {
    return <p className={styles.PreLoader}>Welkom bij Mijn Amsterdam</p>;
  }

  const dialogTimeoutSettings = {
    secondsBeforeDialogShow:
      validityInSeconds ||
      DefaultAutologoutDialogSettings.secondsBeforeDialogShow,
  };

  // Render the main app only if we are authenticated
  return isAuthenticated ? (
    <>
      <AppAuthenticated />
      <AutoLogoutDialog settings={dialogTimeoutSettings} />
    </>
  ) : (
    <AppNotAuthenticated />
  );
}

export default function App() {
  useAnalytics(!!getOtapEnvItem('analyticsId'));
  useScript('/js/usabilla.js', false, true, IS_PRODUCTION);

  const sendToSentry = (error: Error, componentStack: string) => {
    Sentry.captureException(error, {
      extra: {
        componentStack,
      },
    });
  };

  return (
    <RecoilRoot>
      <BrowserRouter>
        <ErrorBoundary
          onError={sendToSentry}
          FallbackComponent={ApplicationError}
        >
          <SessionState>
            <AppLanding />
          </SessionState>
        </ErrorBoundary>
      </BrowserRouter>
    </RecoilRoot>
  );
}
