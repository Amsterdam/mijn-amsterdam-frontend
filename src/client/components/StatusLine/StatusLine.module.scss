@import '../../styles/_global.scss';

.StatusLine {
  margin-bottom: 2rem;

  --clampedGradientBgColor: #f5f5f5;
}

.List {
  @include reset-list-style;
  padding: 1rem 0;

  @include select(DocumentList_DocumentListItem) {
    margin-bottom: 1rem;

    @include select(Button_Icon) {
      margin-left: -17px;
    }
  }

  @include select(DocumentList_DocumentLinkError) {
    @include select(Button_Icon) {
      margin-left: -21px;
    }
  }
}

.ListHeading {
  color: $color-neutral-grey4;
}

.StatusTitle {
  display: block;
}

.StatusDate {
  display: block;
  white-space: nowrap;
  color: $color-neutral-grey4;
}

.StatusConnection {
  position: relative;
  width: 3.4rem;

  @include mq-tablet() {
    position: absolute;
    height: 100%;
    left: 2.6rem;
  }

  @include mq-phone() {
    left: 1rem;
  }
}

.ConnectLine {
  display: block;
  width: 0.2rem;
  top: -0.1rem; // overlapping border-top of different color
  bottom: -0.1rem; // overlapping border-top of different color
  position: absolute;
  left: 50%;
  background: $color-neutral-grey3;

  &--checked {
    background: $color-primary-darkblue;
  }

  &--start {
    background: linear-gradient(
      to bottom,
      transparent 0%,
      transparent 3rem,
      $color-neutral-grey3 3rem,
      $color-neutral-grey3 100%
    );
  }

  &--start-checked {
    background: linear-gradient(
      to bottom,
      transparent 0%,
      transparent 3rem,
      $color-primary-darkblue 3rem,
      $color-primary-darkblue 100%
    );
  }

  &--start-faded {
    background: linear-gradient(
      to bottom,
      $color-neutral-grey1 0%,
      $color-neutral-grey1 3rem,
      $color-neutral-grey3 3rem,
      $color-neutral-grey1 100%
    );
  }

  &--middle-faded {
    background: linear-gradient(
      to bottom,
      $color-primary-darkblue 0%,
      $color-neutral-grey3 75%,
      $color-neutral-grey1 100%
    );
  }

  &--middle-checked-next {
    background: linear-gradient(
      to bottom,
      $color-primary-darkblue 0%,
      $color-primary-darkblue 3rem,
      $color-neutral-grey3 3rem,
      $color-neutral-grey3 100%
    );
  }

  &--end {
    background: linear-gradient(
      to bottom,
      $color-neutral-grey3 0%,
      $color-neutral-grey3 3rem,
      transparent 0%
    );
  }

  &--end-active,
  &--end-checked {
    background: linear-gradient(
      to bottom,
      $color-primary-darkblue 0%,
      $color-primary-darkblue 3rem,
      transparent 0%
    );
  }
}

.Checkmark {
  display: block;
  position: relative;
  border-radius: 50%;
  left: -1.1rem;
  color: $color-white;
  text-align: center;
  font-size: 0.8em;
  font-weight: 600;
  line-height: 1.5;
  width: 2rem;
  height: 2rem;
  top: 2rem;
  left: calc(50% - 0.9rem);

  background-color: $color-neutral-grey3;
  background-repeat: no-repeat;
  background-position: center;
  background-size: 0.9em 0.9em;

  &--active,
  &--checked {
    background-color: $color-primary-darkblue;
    background-image: url('../../assets/icons/Checkmark__white.svg');
  }

  &--active {
    width: 3.2rem;
    height: 3.2rem;
    left: calc(50% - 1.5rem);
    top: 1.4rem;
  }
}

.ListItem {
  &--highlight {
    background-color: $color-neutral-grey1;

    &,
    + .ListItem {
      @include select(StatusLine_ListItemInner) {
        border-top: 0;
      }
    }
  }

  &Inner {
    display: flex;
    padding: 0 7.5rem 0 6.2rem;
    border-top: 1px solid $color-neutral-grey2;

    @include mq-tablet() {
      display: block;
      position: relative;
      padding: 0 1rem 0 6.8rem;
    }

    @include mq-phone() {
      padding: 0 1rem 0 4rem;
    }
  }

  .Panel {
    position: relative;
    padding: 1.8rem;
    flex: 1;

    &--documents {
      width: auto;
      overflow: hidden;
    }

    &--status {
      padding: 1.8rem 0 1.8rem 6.5rem;
    }

    @include mq-tablet() {
      &,
      &--status {
        padding: 1.8rem 1rem 1rem 3rem;
        position: static;
      }
    }

    &--description {
      @include media-q('>tablet') {
        flex: 0 0 38rem;
      }

      @include mq-tablet {
        padding-top: 0;
      }
    }
  }

  .PanelContent {
    > p {
      margin: 0 0 2rem;

      a {
        @extend .ams-link!optional;
        @extend .ams-link--inline!optional;
      }
    }
  }
}

.NoStatusItems {
  padding: 0;
}

.MoreStatus {
  position: relative;
  background-color: transparent;
  display: flex;
  align-content: center;
  justify-content: center;
  border: 0;
  left: 50%;
  transform: translateX(-50%);
  padding: 0 0.4rem 0 0;
  font-weight: 700;

  svg {
    height: 15px;
    margin-top: 0.3rem;
    transform: rotate(90deg);
  }

  &Closed {
    svg {
      transform: rotate(-90deg);
    }
  }
}

.altDocumentContent {
  display: block;

  > * {
    margin-top: 0;
  }
}
