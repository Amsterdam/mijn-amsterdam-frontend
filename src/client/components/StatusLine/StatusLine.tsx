import { Heading } from '@amsterdam/design-system-react';
import classnames from 'classnames';
import { CSSProperties } from 'react';
import { defaultDateFormat } from '../../../universal/helpers/date';
import { ComponentChildren } from '../../../universal/types';
import {
  AltDocumentContent,
  GenericDocument,
  StatusLineItem,
} from '../../../universal/types/App.types';
import DocumentList from '../DocumentList/DocumentList';
import InnerHtml from '../InnerHtml/InnerHtml';
import styles from './StatusLine.module.scss';

interface StatusLinePanelProps {
  children: ComponentChildren;
  name?: string;
}

export function StatusLinePanel({ children, name }: StatusLinePanelProps) {
  const classNames = classnames(styles.Panel, name && styles[`Panel--${name}`]);
  return <div className={classNames}>{children}</div>;
}

interface StatusLinePanelStatusProps {
  datePublished?: string;
  status: string;
}

export function StatusLinePanelStatus({
  datePublished,
  status,
}: StatusLinePanelStatusProps) {
  return (
    <StatusLinePanel name="status">
      <strong className={styles.StatusTitle}>{status}</strong>
      {!!datePublished && (
        <time className={styles.StatusDate}>
          {defaultDateFormat(datePublished)}
        </time>
      )}
    </StatusLinePanel>
  );
}

interface StatusLinePanelDescriptionProps {
  content?: string;
}

export function StatusLinePanelDescription({
  content = '',
}: StatusLinePanelDescriptionProps) {
  return (
    <StatusLinePanel name="description">
      <InnerHtml className={styles.PanelContent}>{content}</InnerHtml>
    </StatusLinePanel>
  );
}

interface StatusLinePanelDocumentsProps {
  documents: GenericDocument[];
  altDocumentContent?: AltDocumentContent;
  trackPath?: (document: GenericDocument) => string;
}

export function StatusLinePanelDocuments({
  documents = [],
  altDocumentContent,
  trackPath,
}: StatusLinePanelDocumentsProps) {
  return (
    <StatusLinePanel name="documents">
      {!!altDocumentContent && typeof altDocumentContent === 'string' ? (
        <InnerHtml el="span" className={styles.altDocumentContent}>
          {altDocumentContent}
        </InnerHtml>
      ) : (
        altDocumentContent
      )}
      {!!documents?.length && (
        <DocumentList documents={documents} trackPath={trackPath} />
      )}
    </StatusLinePanel>
  );
}

interface StatusLineItemProps {
  children: ComponentChildren;
  highlight?: boolean;
  style?: CSSProperties;
}

export function LineItem({
  children,
  highlight = false,
  style,
}: StatusLineItemProps) {
  return (
    <li
      style={style ? style : {}}
      className={classnames(
        styles.ListItem,
        highlight && styles['ListItem--highlight']
      )}
    >
      <div className={styles.ListItemInner}>{children}</div>
    </li>
  );
}

function StatusLineConnectionPlaceholder() {
  return <div className={styles.StatusConnection} />;
}

interface StatusLineConnectionProps {
  index: number;
  total: number;
  isActive: boolean;
  isChecked: boolean;
}

function StatusLineConnection({
  index,
  total,
  isChecked,
  isActive,
}: StatusLineConnectionProps) {
  const maxIndex = total - 1;
  const isEnd = maxIndex !== -1 ? index === maxIndex : false;

  let status = null;
  let ariaLabel = 'Toekomstige status, nog niet aangevinkt.';
  switch (true) {
    case isActive:
      ariaLabel = 'Huidige status, aangevinkt.';
      break;
    case isChecked:
      ariaLabel = 'Status aangevinkt.';
      break;
  }

  if (total === 1) {
    status = (
      <span
        className={classnames(
          styles.Checkmark,
          isActive && styles['Checkmark--active'],
          isChecked && styles['Checkmark--checked']
        )}
      />
    );
  }

  if (total > 1) {
    if (index === 0) {
      status = (
        <>
          <span
            className={classnames(
              styles.ConnectLine,
              isChecked && !isActive && total > 1
                ? styles['ConnectLine--start-checked']
                : styles['ConnectLine--start']
            )}
          />
          <span
            className={classnames(
              styles.Checkmark,
              isActive && styles['Checkmark--active'],
              isChecked && styles['Checkmark--checked']
            )}
          />
        </>
      );
    } else if (!isEnd && isActive) {
      status = (
        <>
          <span
            className={classnames(
              styles.ConnectLine,
              styles['ConnectLine--middle-checked-next']
            )}
          />
          <span
            className={classnames(
              styles.Checkmark,
              isActive && styles['Checkmark--active'],
              isChecked && styles['Checkmark--checked']
            )}
          />
        </>
      );
    } else if (isEnd) {
      status = (
        <>
          <span
            className={classnames(
              styles.ConnectLine,
              isActive && styles['ConnectLine--end-active'],
              isChecked
                ? styles['ConnectLine--end-checked']
                : styles['ConnectLine--end']
            )}
          />
          <span
            className={classnames(
              styles.Checkmark,
              isActive && styles['Checkmark--active'],
              isChecked && styles['Checkmark--checked']
            )}
          />
        </>
      );
    } else {
      status = (
        <>
          <span
            className={classnames(
              styles.ConnectLine,
              isChecked && styles['ConnectLine--checked']
            )}
          />
          <span
            className={classnames(
              styles.Checkmark,
              isActive && styles['Checkmark--active'],
              isChecked && styles['Checkmark--checked']
            )}
          />
        </>
      );
    }
  }
  return (
    <div aria-label={ariaLabel} className={styles.StatusConnection}>
      {status}
    </div>
  );
}

interface StatusLineProps {
  items: StatusLineItem[];
  // Deprecated
  trackCategory?: string;
  altDocumentContent?: AltDocumentContent;
  // Deprecated
  id?: string;
  statusLabel?: string;
  showStatusLineConnection?: boolean;
  className?: string;
  documentPathForTracking?: (document: GenericDocument) => string;
}

export default function StatusLine({
  items,
  statusLabel = 'Status',
  className,
  showStatusLineConnection = true,
  documentPathForTracking,
}: StatusLineProps) {
  return (
    <div className={classnames(styles.StatusLine, className)}>
      <Heading level={4} size="level-4" className={styles.ListHeading}>
        {statusLabel}
      </Heading>
      {!!items.length && (
        <ul className={styles.List}>
          {items.map((item, index) => (
            <LineItem
              key={`step-${item.id}-${index}`}
              highlight={item.isActive}
            >
              {showStatusLineConnection === true ? (
                <StatusLineConnection
                  index={index}
                  total={items.length}
                  isActive={item.isActive}
                  isChecked={item.isChecked}
                />
              ) : (
                <>
                  <StatusLineConnectionPlaceholder />
                </>
              )}

              <StatusLinePanelStatus
                datePublished={item.datePublished}
                status={item.status}
              />

              <StatusLinePanelDescription content={item.description} />

              <StatusLinePanelDocuments
                documents={item.documents ?? []}
                altDocumentContent={item.altDocumentContent}
                trackPath={documentPathForTracking}
              />
            </LineItem>
          ))}
        </ul>
      )}
      {!items.length && (
        <p className={styles.NoStatusItems}>Er is geen status beschikbaar.</p>
      )}
    </div>
  );
}
