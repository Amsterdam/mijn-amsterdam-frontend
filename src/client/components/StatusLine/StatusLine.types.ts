import { GenericDocument } from '../../../universal/types';

export type StepType =
  | 'first-step'
  | 'last-step'
  | 'intermediate-step'
  | 'single-step';

export type AltDocumentContent = string | JSX.Element;

export interface StatusLineItem {
  id: string;
  status: string;
  datePublished: string;
  description: string;
  documents: GenericDocument[];
  isActive?: boolean;
  isChecked?: boolean;
  isHighlight?: boolean;
  altDocumentContent?: AltDocumentContent;
  [key: string]: any;
}
