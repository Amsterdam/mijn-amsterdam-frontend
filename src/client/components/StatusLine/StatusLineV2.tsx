import { ReactNode } from 'react';

import {
  Grid,
  Heading,
  Icon,
  OrderedList,
} from '@amsterdam/design-system-react';
import { CheckmarkIcon } from '@amsterdam/design-system-react-icons';
import classNames from 'classnames';

import styles from './StatusLineV2.module.scss';
import { defaultDateFormat } from '../../../universal/helpers/date';
import { GenericDocument, StatusLineItem } from '../../../universal/types';
import { DocumentLink } from '../DocumentList/DocumentLink';
import InnerHtml from '../InnerHtml/InnerHtml';

interface StatusStepDocumentsProps {
  documents?: GenericDocument[];
  altDocumentContent?: ReactNode;
}

export function StatusStepDocuments({
  documents = [],
  altDocumentContent,
}: StatusStepDocumentsProps) {
  return (
    <>
      {!!altDocumentContent && typeof altDocumentContent === 'string' ? (
        <InnerHtml el="span">{altDocumentContent}</InnerHtml>
      ) : (
        altDocumentContent
      )}
      {!!documents?.length && (
        <OrderedList className={styles.StepDocumentsList} markers={false}>
          {documents.map((document) => (
            <OrderedList.Item key={document.id}>
              <DocumentLink key={document.id} document={document} />
            </OrderedList.Item>
          ))}
        </OrderedList>
      )}
    </>
  );
}

type StatusIndicationProps = {
  step: StatusLineItem;
};

function StatusIndication({ step }: StatusIndicationProps) {
  let content: ReactNode = '';
  let ariaLabel: string = 'Toekomstige status';

  if (step.isChecked && !step.isActive) {
    ariaLabel = 'Status Afgerond';
    content = (
      <Icon
        size="level-6"
        className={styles.StatusIndicationCheckmark}
        aria-label="Status Afgerond"
        svg={CheckmarkIcon}
      />
    );
  } else if (step.isActive) {
    ariaLabel = 'Huidige status';
  }

  return (
    <span aria-label={ariaLabel} className={styles.StepStatusIndication}>
      {content}
    </span>
  );
}

type StepsProps = {
  steps: StatusLineItem[];
  title?: string;
};

export function Steps({ steps, title }: StepsProps) {
  return (
    <Grid className={styles.Steps}>
      {title && (
        <Grid.Cell start={2} span={10}>
          <Heading>{title}</Heading>
        </Grid.Cell>
      )}
      <Grid.Cell start={2} span={10}>
        <OrderedList className={styles.StepsList} markers={false}>
          {steps.map((item) => (
            <OrderedList.Item
              className={classNames(
                styles.Step,
                item.isChecked && !item.isActive && styles['Step--checked'],
                item.isActive && styles['Step--active']
              )}
              key={item.id + item.datePublished}
            >
              <Heading className={styles.StepStatus} level={4}>
                <StatusIndication step={item} />
                {item.status}
              </Heading>
              <time
                className={styles.StepStatusDate}
                dateTime={item.datePublished}
              >
                {defaultDateFormat(item.datePublished)}
              </time>
              {item.description && <InnerHtml>{item.description}</InnerHtml>}
              {!!(item.altDocumentContent || item.documents?.length) && (
                <StatusStepDocuments
                  documents={item.documents}
                  altDocumentContent={item.altDocumentContent}
                />
              )}
            </OrderedList.Item>
          ))}
        </OrderedList>
      </Grid.Cell>
    </Grid>
  );
}
