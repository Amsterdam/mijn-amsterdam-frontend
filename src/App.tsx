import React from 'react';
import { BrowserRouter, Route, Redirect, Switch } from 'react-router-dom';
import useReactRouter from 'use-react-router';

import AppState, { SessionState } from './AppState';
import { AppRoutes } from './App.constants';
import MainHeader, {
  MainHeaderProps,
} from './components/MainHeader/MainHeader';
import MainFooter from './components/MainFooter/MainFooter';
import Profile from './pages/Profile/Profile';
import NotFound from './pages/NotFound/NotFound';
import styles from './App.module.scss';
import MyUpdates from 'pages/MyUpdates/MyUpdates';
import MyTips from 'pages/MyTips/MyTips';
import Dashboard from 'pages/Dashboard/Dashboard';
import Landing from 'pages/Landing/Landing';
import Burgerzaken from 'pages/Burgerzaken/Burgerzaken';
import Belastingen from 'pages/Belastingen/Belastingen';
import Jeugdhulp from 'pages/Jeugdhulp/Jeugdhulp';
import Wonen from 'pages/Wonen/Wonen';
import Inkomen from 'pages/Inkomen/Inkomen';
import Zorg from 'pages/Zorg/Zorg';
import ZorgDetail from 'pages/ZorgDetail/ZorgDetail';
import InkomenDetail from 'pages/InkomenDetail/InkomenDetail';
import MyArea from 'pages/MyArea/MyArea';

function MainApp({ person, isAuthenticated }: MainHeaderProps) {
  const { location } = useReactRouter();

  return location.pathname === AppRoutes.MY_AREA ? (
    <MyArea />
  ) : (
    <>
      <MainHeader person={person} isAuthenticated={isAuthenticated} />
      <div className={styles.App}>
        <Switch>
          <Route exact path={AppRoutes.ROOT} component={Dashboard} />
          <Redirect from={AppRoutes.API_LOGIN} to={AppRoutes.ROOT} />
          <Route path={AppRoutes.MY_UPDATES} component={MyUpdates} />
          <Route path={AppRoutes.PROFILE} component={Profile} />
          <Route path={AppRoutes.MY_TIPS} component={MyTips} />
          <Route path={AppRoutes.BURGERZAKEN} component={Burgerzaken} />
          <Route path={AppRoutes.BELASTINGEN} component={Belastingen} />
          <Route path={AppRoutes.JEUGDHULP} component={Jeugdhulp} />
          <Route path={AppRoutes.WONEN} component={Wonen} />
          <Route path={`${AppRoutes.INKOMEN}/:id`} component={InkomenDetail} />
          <Route path={AppRoutes.INKOMEN} component={Inkomen} />
          <Route path={`${AppRoutes.ZORG}/:id`} component={ZorgDetail} />
          <Route path={AppRoutes.ZORG} component={Zorg} />
          <Route path={AppRoutes.MY_AREA} component={MyArea} />
          <Route component={NotFound} />
        </Switch>
      </div>
      <MainFooter />
    </>
  );
}

export default function App() {
  return (
    <BrowserRouter>
      <SessionState
        render={session => {
          if (session.isLoading) {
            return <p>...</p>;
          }
          // Render the main app only if we are authenticated
          return session.isAuthenticated ? (
            <AppState
              session={session}
              render={({ SESSION, BRP }) => (
                <MainApp
                  person={BRP.person}
                  isAuthenticated={SESSION.isAuthenticated}
                />
              )}
            />
          ) : (
            <>
              <MainHeader />
              <Landing />
              <MainFooter />
            </>
          );
        }}
      />
    </BrowserRouter>
  );
}
