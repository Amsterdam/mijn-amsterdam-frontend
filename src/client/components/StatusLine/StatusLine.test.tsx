import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import StatusLine from './StatusLine';
import { StatusLineItem } from '../../../universal/types/App.types';

const TEST_ITEM: StatusLineItem = {
  id: 'step-1',
  status: 'FOO',
  datePublished: '2020-01-10',
  description: 'A line item',
  documents: [
    {
      title: 'Document title',
      id: 'document-1',
      url: 'https://foo.bar/document.pdf',
      datePublished: '2020-01-10',
    },
  ],
  isActive: false,
  isChecked: true,
};

const items: StatusLineItem[] = [TEST_ITEM];

describe('<StatusLine />', () => {
  it('Renders correctly', () => {
    render(
      <RecoilRoot>
        <BrowserRouter>
          <StatusLine id="unittest" trackCategory="unittest" items={items} />
        </BrowserRouter>
      </RecoilRoot>
    );
    expect(screen.getByText(TEST_ITEM.description!)).toBeInTheDocument();
    expect(screen.getByText(TEST_ITEM.status)).toBeInTheDocument();
    expect(
      screen.getByText(TEST_ITEM?.documents![0]?.title)
    ).toBeInTheDocument();
  });
});
