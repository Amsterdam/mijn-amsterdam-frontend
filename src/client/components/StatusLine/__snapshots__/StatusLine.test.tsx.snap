// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusLine /> Renders the correct html 1`] = `"<div class=\\"StatusLine\\"><h4 class=\\"Heading Heading__tiny ListHeading\\">Status</h4><ul class=\\"List\\"><li style=\\"display:none\\" class=\\"ListItem\\"><div class=\\"ListItemInner\\"><div class=\\"StatusConnection\\"><span class=\\"ConnectLine ConnectLine--start\\"></span><span class=\\"Checkmark Checkmark--checked\\"></span></div><div class=\\"Panel Panel--status\\"><strong class=\\"StatusTitle\\">FOO</strong><time class=\\"StatusDate\\">10 januari 2020</time></div><div class=\\"Panel Panel--description\\"><div class=\\"PanelContent\\">A line item</div></div><div class=\\"Panel Panel--documents\\"><ul class=\\"DocumentList\\"><li class=\\"DocumentListItem\\"><a rel=\\"external noopener noreferrer\\" href=\\"https://foo.bar/document.pdf\\" download=\\"Document title\\" class=\\"Button Button__plain Button__lean Linkd DocumentLink download\\"><svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" fill=\\"#000\\" class=\\"Icon Icon__left\\">Download.svg</svg>Document title</a></li></ul></div></div></li></ul></div>"`;
